Classes/common/CDTDatastore.m

```diff
@@ -289,6 +289,41 @@ - (NSArray *)getAllDocuments
     return result;
 }
 
+- (NSArray *)getAllDocumentIds
+{
+    if (![self ensureDatabaseOpen]) {
+        return nil;
+    }
+    
+    NSArray *result = [NSArray array];
+    struct TDQueryOptions query = {.limit = UINT_MAX,
+                                   .inclusiveEnd = YES,
+                                   .skip = 0,
+                                   .descending = NO,
+                                   .includeDocs = NO};
+    
+    BOOL done = NO;
+    do {
+        NSMutableArray *batch = [NSMutableArray array];
+        
+        NSDictionary *dictResults = [self.database getDocsWithIDs:nil options:&query];
+        
+        for (NSDictionary *row in dictResults[@"rows"]) {
+            [batch addObject:row[@"id"]];
+        }
+        
+        result = [result arrayByAddingObjectsFromArray:batch];
+        
+        done = ((NSArray *)dictResults[@"rows"]).count == 0;
+        
+        query.skip = query.skip + query.limit;
+        
+    } while (!done);
+    
+    return result;
```

Comment:
I copied this logic from `getAllDocuments` which uses `done` to manage the `do...while` loop.  It is essentially the same logic.  I made the change suggested by @gadamc in `getAllDocumentIds`.  Let me know if I should update the `getAllDocuments` logic as well, as part of this PR, open an FB case for the work, or simply leave it alone.

Also, in order to remove `done` from the `do...while`, `dictResults` needs to be declared outside of the loop.  This is done in 45e17aa70c46c0a27263162d7bf9b9e64a31f29b.